design-proposals/dra.md

```diff
@@ -0,0 +1,529 @@
+	// +optional
+	DeviceName *string `json:"deviceName,omitempty"`
+	// DeviceAttributes are the attributes published by the driver running on the node in
+	// resourceslice.spec.devices.basic.attributes. The attributes are distinguished by deviceName
+	// and resourceclaim.spec.devices.requests.deviceClassName.
+	// +optional
+	DeviceAttributes map[string]DeviceAttribute `json:"deviceAttributes,omitempty"`
+}
+```
+
+The first section vmi.spec.resourceClaims will have a list of devices needed to be allocated for the VM. Having this
+available as a list will allow users to use the device from this list in GPU section or HostDevices section of the 
+DomainSpec API. 
+
+In v1alpha3 version of [DRA API](https://pkg.go.dev/k8s.io/api@v0.31.0/resource/v1alpha3#DeviceClaim), multiple drivers
+could potentially provision devices that are part of a single claim. For this reason, a separate list of claims required
+for the VMI (section 1) is needed instead of mentioning the resource claim in devices section 
+([see Alternate Designs](#Alternative 1))
+
+The second sections allows for the resource claim to be used in the spec.domain.devices section. The two uses cases
+currently handled by the design are:
+
+1. allowing the devices to be used as a gpu device (spec.domain.devices.gpu)
+2. allowing the devices to be used as a host device (spec.domain.device.hostDevices)
+
+The status section of the VMI will contain information of the allocated devices for the VMI when the information is
+available in DRA APIs. The same information will be accessible in virt-handler, virt-launcher and sidecar containers. 
+This allows for device information to flow from DRA APIs into KubeVirt stack. 
+
+The virt-launcher will have the logic of converting a GPU device into its corresponding domxml. For use-cases that are 
+not handled in-tree, a sidecar container could be envisioned which will convert the information available in status to 
+the corresponding domxml.
+
+### Examples
+
+### VM API with PassThrough GPU
+
+```
+---
+# this is a cluster scoped resource
+apiVersion: resource.k8s.io/v1alpha3
+kind: DeviceClass
+metadata:
+  name: gpu.example.com
+spec:
+  selectors:
+  - cel:
+      expression: device.driver == 'gpu.example.com'
+---
+apiVersion: resource.k8s.io/v1alpha3
+kind: ResourceClaimTemplate
+metadata:
+  name: pgpu-claim-template
+spec:
+  spec:
+    devices:
+      requests:
+        - name: pgpu-request-name
+          deviceClassName: gpu.example.com
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  labels:
+    kubevirt.io/vm: vm-cirros
+  name: vm-cirros
+spec:
+  resourceClaims:
+  - name: pgpu-claim-name
+    source:
+      resourceClaimTemplateName: pgpu-claim-template
+  domain:
+    devices:
+      gpus:
+      - name: pgpu
+        claim: 
+          name: pgpu-claim-name
+          request: pgpu-request-name
```

Comment:
I don't quite understand why we cannot simply reference the name of the resourceClaim, it is done in the same way for the volumes/disks. Why do we need an additional level with the name and the request?